smurfs/src/components/App.js

```diff
@@ -1,22 +1,88 @@
 import React, { Component } from 'react';
 import './App.css';
+import { connect } from 'react-redux';
+import { getSmurfs, addSmurf, deleteSmurf } from '../actions'
 /*
  to wire this component up you're going to need a few things.
  I'll let you do this part on your own. 
  Just remember, `how do I `connect` my components to redux?`
  `How do I ensure that my component links the state to props?`
  */
 class App extends Component {
+  state = {
+    name: '',
+    age: '',
+    height: ''
+  }
+
+  componentDidMount() {
+    this.props.getSmurfs()
+  }
+
+  handleChanges = (e) => {
+    this.setState({
+      [e.target.name]: e.target.value
+    })
+  }
+
+  addSmurf = (e) => {
+    e.preventDefault();
+    this.props.addSmurf(this.state);
+    this.setState({
+      name: '',
+      age: '',
+      height: ''
+    })
+  }
+
+  deleteSmurf = (e, id) => {
+    console.log(id)
+    this.props.deleteSmurf(id)
+  }
+
   render() {
     return (
       <div className="App">
-        <h1>SMURFS! 2.0 W/ Redux</h1>
-        <div>Welcome to your Redux version of Smurfs!</div>
-        <div>Start inside of your `src/index.js` file!</div>
-        <div>Have fun!</div>
+        <nav>
+          <div className="nav-wrapper #2196f3 blue">
+            <a href="#" className="brand-logo center">Smurf Database</a>
+          </div>
+        </nav>
+        <div className="container" style={{margin: "40px auto"}}>
+          <form onSubmit={this.addSmurf}>
+            <input placeholder="Name" value={this.state.name} onChange={this.handleChanges} name="name"/>
+            <input placeholder="Age" value={this.state.age} onChange={this.handleChanges} name="age" />
+            <input placeholder="Height" value={this.state.height} onChange={this.handleChanges} name="height" />
+            <button style={{margin: "40px auto"}} className="waves-effect #2196f3 blue btn">Add Smurf</button>
+          </form>
+          </div>
+          <div className="container" style={{display:"flex", flexWrap:"wrap-reverse"}}>
+            {this.props.smurfs.map( (smurf, index) => (
+              <div >
```

Comment:
The key, I think, should go on this line 61 div here. It *should* stop the console warning.